cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+def installation_allowed?
+  %w[bionic focal xenial].include?(node['lsb']['codename']) && node['kernel']['machine'] == 'amd64'
+end
+
+if installation_allowed?
+  def deb_url
+    base_url = node['travis_build_environment']['ibm_advanced_tool_chain_apt_deb_url']
+    "#{base_url} #{node['lsb']['codename']} at#{node['travis_build_environment']['ibm_advanced_tool_chain_version']}"
+  end
+
+  execute "apt-key-add-ibm-advanced-tool-chain" do
+    command "wget --auth-no-challenge -qO - #{node['travis_build_environment']['ibm_advanced_tool_chain_apt_key_url']} | apt-key add -"
+    action :nothing
+  end
+
+  execute "apt-add-sources-list" do
```

Comment:
Resource `apt_repository` could be used here

Reply:
Done